server/controllers/importers/graphml.js

```diff
@@ -1,19 +1,84 @@
 var semanticChecker = require(__dirname + "/../semantic-checker");
 
+var graphmlWarnings = {
+  EDGES_WITHOUT_WEIGHTS: {
+    warningCode: "EDGES_WITHOUT_WEIGHTS",
+    errorDescription: "GRNsight has detected that one or more edges in your network are missing numerical weight" +
+    " values. Because the algorithm GRNsight uses for determining the arrowhead type and the color and thickness of" +
+    " the edges requires numerical weight values, your graph will display as an unweighted graph with black edges" +
+    " and pointed arrowheads. If you want to display the network as a weighted graph, please modify your input file" +
+    " to include weight values for all edges."
+  },
+
+  EDGE_DEFAULT_NOT_DIRECTED: {
+    warningCode: "EDGE_DEFAULT_NOT_DIRECTED",
+    errorDescription: "GRNsight interprets the graph as directed unconditionally."
+  }
+};
+
+var graphmlErrors = {
+  GRAPHML_GENERAL_SYNTAX_ERROR: function (error) {
+    return {
+      errorCode: "GRAPHML_GENERAL_SYNTAX_ERROR",
+      possibleCause: "There are a number of things that could've triggered this error, but the general gist is that" +
+      " there is something syntactically wrong with your file. The parcer we are using has associated your syntax error" +
+      " with this message: " + error + ".",
+      suggestedFix: "Please check the format of your file and make sure that it is in line with our Documentation page." +
+      " Some common errors to check for are missing start/end tags, missing quotation marks, and proper spelling of" +
+      " all attribute names."
+    }
+  }
+};
+
 module.exports = function (graphml) {
   var graph, key;
 
+  var network = {
+    genes: [],
+    links: [],
+    errors: [],
+    warnings: [],
+    positiveWeights: [],
+    negativeWeights: [],
+    sheetType: constants.UNWEIGHTED
+  };
+
   // Note this relies on sync execution being the default, *not* async.
   //
   // Limitation is due to the way the import function is expected to return its result.
   // To address this later on, import functions should accept a callback instead.
   parseString(graphml, function (err, result) {
-    key = result.graphml && result.graphml.key;
-    graph = result.graphml && result.graphml.graph && result.graphml.graph[0];
+    if (err) {
+      pushRelevantError(err);
+    } else {
+      key = result.graphml && result.graphml.key;
+      graph = result.graphml && result.graphml.graph && result.graphml.graph[0];
+    }
   });
 
+  if (network.errors.length > 0) { return semanticChecker(network); }
+
+  function pushRelevantError(err) {
+    var parseError = readErrorFromErr(err);
+    network.errors.push(graphmlErrors.GRAPHML_GENERAL_SYNTAX_ERROR(parseError));
+    // switch(parseError) {
+    //   case "Invalid attribute name":
+    //     network.errors.push(graphmlErrors.UNKNOWN_ERROR);
+    //     break;
+    //   default:
+    //     network.errors.push(graphmlErrors.GRAPHML_GENERAL_SYNTAX_ERROR);
+    // }
```

Comment:
Add a comment here for next semester:

    // TODO Ask Dondi about how to do this in a more data-driven manner.